recipe_scrapers/panelinha.py

```diff
@@ -26,10 +30,24 @@ def instructions(self):
         instructions = self.soup.find(
             "h4", string="Modo de preparo"
         ).nextSibling.findAll("li")
-
-        return "\n".join(
-            [normalize_string(instruction.get_text()) for instruction in instructions]
-        )
+        instructions = [
+            normalize_string(instruction.get_text()) for instruction in instructions
+        ]
+
+        # Some recipes pages have a different html structure.
+        if not instructions:
+            instructions = self.soup.find(
+                "h4", string="Modo de preparo"
+            ).nextSibling.p.contents
+            instructions = filter(lambda x: isinstance(x, str), instructions)
+            instructions = [
```

Comment:
Also, this comprehension could be made into a generator by replacing the square brackets with parenthesis so that we don't have an intermediary object. Might reduce speed, but at least we don't store multiple large arrays.

This is still a nit. Not a big issue.